pkg/scheduler/framework/runtime/framework_test.go

```diff
@@ -1704,7 +1709,7 @@ func TestPreBindPlugins(t *testing.T) {
 					inj:  injectedResult{PreBindStatus: int(framework.UnschedulableAndUnresolvable)},
 				},
 			},
-			wantStatus: framework.AsStatus(fmt.Errorf(`running PreBind plugin "TestPlugin": %w`, errInjectedStatus)),
+			wantStatus: framework.NewStatus(framework.UnschedulableAndUnresolvable, injectReason).WithFailedPlugin("TestPlugin"),
```

Comment:
this is a behavior change, it's changing the status from `Error` to `UnschedulableAndUnresolvable`. This needs to be discussed.

Reply:
Yes, discussion is always needed. 

Currently, after we run `RunBindPlugins` and `RunPreBindPlugins`, we will wrap the status and always return a Error one, then even the pod is rejected for unschedulable reason, we'll also put it back to backingOffQ and activeQ rather than unschedulablePods. See [`MoveAllToActiveOrBackoffQueue`](https://github.com/kubernetes/kubernetes/blob/5debf63a742fab8f3d9b3672a1fd3f14d2289dce/pkg/scheduler/internal/queue/scheduling_queue.go#L626-L636), but it will fail again. So we should distinguish these status apart.

Reply:
cc @ahg-g @alculquicondor @Huang-Wei

Reply:
The only in-tree plugin that we have uses Error, so I'm fine with this change.

Reply:
This is fine even if bind was returning unschedulable.

Reply:
Fine with this change to honor the status returned by plugins. Also tagged @cofyc above to see he wants to change Error to Unschedulable so we don't immediately retry pods that failed on PreBind of volumebinding. (w/o this PR, no matter what status is returned from PreBind/Bind, it's treated as Error)